contracts/forwarder/Eip712Forwarder.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import "./IForwarder.sol";
+// TODO now: move getError to the forwarder folder
+import "../utils/GsnUtils.sol";
+
+contract Eip712Forwarder is IForwarder {
+    using ECDSA for bytes32;
+
+    string public constant GENERIC_PARAMS = "address to,bytes data,address from,uint256 nonce,uint256 gas";
```

Comment:
I would remove gas from the forwarder. This can be handled by forwarder extension.
This is important because EVM is not great for gas handling right now